src/opensynth/models/faraday/gaussian_mixture/metrics.py

```diff
@@ -40,7 +40,9 @@ def update(self, responsibilities: torch.Tensor) -> None:
         self.responsibilities.add_(responsibilities.sum(0))
 
     def compute(self) -> torch.Tensor:
-        return self.responsibilities / self.responsibilities.sum()
+        return (self.responsibilities / self.responsibilities.sum()).add_(
+            1e-16
+        )
```

Comment:
Should this regularisation 1e-6 term be added to the denominator only, to prevent the issue of dividing by zero?